### ISSUE TITLE
Custom Block contextual link

### ISSUE THREAD

**Issue by BWPanda** [backdrop/backdrop-issues#4390]
**Description of the need**

I have a custom block on the homepage with intro text that I want the client to be able to edit. So I gave them permission to 'Use contextual links' and 'Administer custom blocks'. However when they hover over the homepage block, no contextual link appears. It seems that you need the 'Administer Layouts' permission to see/use that, but I didn't/don't want to give them that permission. So they can still edit the block using the admin bar, but it'd be nice if they could use the contextual link too.

**Proposed solution**

It'd be nice if the contextual link was displayed and changed to go to `/admin/structure/block/manage/[BLOCK]/configure` instead of the Layouts UI if they don't have that permission.

**Comment by indigoxela** [comment]
> It'd be nice if the contextual link was displayed and changed to go to /admin/structure/block/manage/[BLOCK]/configure instead of the Layouts UI if they don't have that permission.

Sounds like a good idea, but then again...

Letting the same link (with same link text?) point to different locations, based on permissions, seems like a guaranteed source of misunderstanding between users of different roles.

@BWPanda or did you have something different in mind?

**Comment by BWPanda** [comment]
I think I'd just like to see an easy way for my client to edit their block via a contextual link without needing the Layouts UI permission. However that's solved is fine by me (e.g. a new link added (e.g. 'Edit custom block' (with the existing link renamed to 'Configure block in layout')), an option to set where block edit links should point, etc.).

**Comment by laryn** [comment]
I agree that this is a need. It might be good to also put a warning on the edit page that this block may be used on multiple pages so changes may affect more than just the page they came from.

**Comment by indigoxela** [comment]
I also agree that there's a need for that. I'm just not sure yet, what's the best (cleanest, least confusing) way to do it.

We have two different paths, one for the layout:
`admin/structure/layouts/manage/home/configure-redirect/[block uuid]`
one for the custom (reusable) block:
`admin/structure/block/manage/[machine name]/configure`

They provide slightly different functionality.
While the first allows all sorts of settings (styles, visibility...), the second only provides a way to _edit_ or delete (possibly translate) a custom block.

**Comment by BWPanda** [comment]
Would 'Configure block' and 'Edit block' (for the respective paths above) be too confusing/similar?

**Comment by klonos** [comment]
I agree that this is a legitimate use case that we need to address. I'm not sure how likely it is to cause confusion by having 2 different links, but then again if we can be proactive and provide an intuitive solution, then why not.

I like "Configure block" vs something like "Edit block content" 👍

**Comment by olafgrabienski** [comment]
I also like the two links approach. @BWPanda for the time being, there is a workaround: Not sure if it's a possibility in your case, but you could use a content type with 'hidden path display' instead of custom blocks, and place nodes of this type as blocks of 'existing content'. These blocks have a contextual edit link to edit the respective node.

**Comment by stpaultim** [comment]
I just ran into this same problem. I'm willing to give my site editor permission to edit the layout, however this doesn't seen like it should be necessary. 

Also, it took me a little while to figure out why a site editor that has permission to use contextual links and to edit custom blocks, was unable to do so (until I gave them permission to edit layouts). 

Not sure of the best solution, but wanted to at least offer support for the idea of having some solution.

**Comment by indigoxela** [comment]
FTR: I recommend to use nodes instead of blocks. They ship with a functioning contextual link for all people with permission to edit them. And then there's no need for granting the layout permission to less privileged editors. Since rabbit hole went into core, there's almost no need for custom blocks anymore (IMO). We probably only use them out of habit - in Drupal they're more important.
